pkg/registry/seedmanagement/managedseed/storage/tableconvertor.go

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) 2020 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package storage
+
+import (
+	"context"
+
+	"k8s.io/apimachinery/pkg/api/meta"
+	metatable "k8s.io/apimachinery/pkg/api/meta/table"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	metav1beta1 "k8s.io/apimachinery/pkg/apis/meta/v1beta1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apiserver/pkg/registry/rest"
+
+	"github.com/gardener/gardener/pkg/apis/seedmanagement"
+)
+
+var swaggerMetadataDescriptions = metav1.ObjectMeta{}.SwaggerDoc()
+
+type convertor struct {
+	headers []metav1beta1.TableColumnDefinition
+}
+
+func newTableConvertor() rest.TableConvertor {
+	return &convertor{
+		headers: []metav1beta1.TableColumnDefinition{
+			{Name: "Name", Type: "string", Format: "name", Description: swaggerMetadataDescriptions["name"]},
+			{Name: "Shoot", Type: "string", Format: "name", Description: swaggerMetadataDescriptions["shootName"]},
+			{Name: "Seed", Type: "string", Format: "name", Description: swaggerMetadataDescriptions["name"]},
+		},
+	}
+}
+
+// ConvertToTable converts the output to a table.
+func (c *convertor) ConvertToTable(ctx context.Context, obj runtime.Object, tableOptions runtime.Object) (*metav1beta1.Table, error) {
+	var (
+		err   error
+		table = &metav1beta1.Table{
+			ColumnDefinitions: c.headers,
+		}
+	)
+
+	if m, err := meta.ListAccessor(obj); err == nil {
+		table.ResourceVersion = m.GetResourceVersion()
+		table.SelfLink = m.GetSelfLink()
+		table.Continue = m.GetContinue()
+	} else {
+		if m, err := meta.CommonAccessor(obj); err == nil {
+			table.ResourceVersion = m.GetResourceVersion()
+			table.SelfLink = m.GetSelfLink()
+		}
+	}
+
+	table.Rows, err = metatable.MetaToTableRow(obj, func(obj runtime.Object, m metav1.Object, name, age string) ([]interface{}, error) {
+		var (
+			managedSeed = obj.(*seedmanagement.ManagedSeed)
+			cells       = []interface{}{}
+			shootName   string
+		)
+
+		if managedSeed.Spec.Shoot != nil {
+			shootName = managedSeed.Spec.Shoot.Name
+		}
+
+		cells = append(cells, managedSeed.Name)
+		cells = append(cells, shootName)
+		cells = append(cells, managedSeed.Name)
```

Comment:
This is actually the seed name. Since it's currently always the managed seed name itself, I guess it's not really needed. I will remove it.

Reply:
@danielfoehrKn I would prefer to forbid specifying the Seed name explicitly and always set it to the ManagedSeed name (or a predictable, unique derivative). Otherwise we would have to deal with naming conflicts, etc. Yes, you are right it should be documented and validated, I just haven't got to that point.